### PR TITLE
session: support leader-and-follower for tidb_replica_read

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve? 
Currently there is no option to support load balance for TiKV peers, we can only read from leader or followers, not both.

### What is changed and how it works?
Support leader-and-follower for tidb_replica_read, which is a simple load balance strategy in all peers of a given region.

### Check List 

Tests 
Unit test

Code changes
Has exported variable/fields change

Side effects

Related changes
Need to cherry-pick to release-4.0-beta.

Release note
support leader-and-follower for tidb_replica_read.

### CODE REVIEW

File: store/tikv/region_cache.go [hunk 1]
```diff
@@ -115,6 +115,31 @@ func (r *RegionStore) follower(seed uint32) int32 {
...
+		if r.storeFails[i] != atomic.LoadUint32(&r.stores[i].fail) {
+			continue
+		}
+		candidates = append(candidates, int32(i))
+	}
+
+	if len(candidates) == 0 {
+		return r.workTiKVIdx
+	} else {
+		return int32(seed) % int32(len(candidates))
```
Comment (lysu) [inline store/tikv/region_cache.go]: `stores` value are `[leader, tiflash, tiflash, follower, follower]`
`seed` be 2, `len(candidates)` be 3  and this method will return idx = `2` then `r.getStorePeer` will return a tiflash node in `AnyStorePeer`

Comment (lysu) [reply store/tikv/region_cache.go]: or stores value are `[leader, follower(fail), follower(fail), follower(ok), follower(ok)]` have same problem

Comment (hicqu) [reply store/tikv/region_cache.go]: Fixed. PTAL.

**Comment by tiancaiamao** [comment]
LGTM

File: kv/kv.go [hunk 2]
```diff
@@ -93,13 +93,14 @@ const (
 	ReplicaReadLeader ReplicaReadType = 1 << iota
 	// ReplicaReadFollower stands for 'read from follower'.
 	ReplicaReadFollower
-	// ReplicaReadLearner stands for 'read from learner'.
-	ReplicaReadLearner
+	// ReplicaReadMixed stands for 'read from leader and follower and learner'.
```
Comment (zhangjinpeng1987) [inline kv/kv.go]: @lysu @hicqu Seems the comment is not correct.

**Pull request merged**